### PR TITLE
Enhancing the documentation: assumptions section

### PR DESCRIPTION
**Pull request by sandrain**
This PR enhances some contents in Assumptions section in the documentation:

- discussing the unifyfs data staging functionalities
- fix the code example of lamination
- clarify some sentences

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Testing (addition of new tests or update to current tests)
- [x] Documentation (a change to man pages or other documentation)

### CODE REVIEW

File: docs/assumptions.rst [hunk 1]
```diff
@@ -95,32 +119,32 @@ Behavior after lamination:
...
 
-    - The file system exists on node local storage only and is not persisted to 
-      stable storage like a parallel file system (PFS). Can be coupled with
+The additional behavior of UnifyFS can be summarized as follows.
 
-    - SymphonyFS or high level I/O or checkpoint library (VeloC) to move data to
-      PFS periodically, or data can be moved manually
+    - UnifyFS exists on node local storage only and is not automatically
+      persisted to stable storage like a parallel file system (PFS). When the
+      data need to be persisted to an external file system, users can use
```
Comment (CamStan) [inline docs/assumptions.rst:126]:
```suggestion
      data needs to be persisted to an external file system, users can use
```

File: docs/assumptions.rst [hunk 2]
```diff
@@ -95,32 +119,32 @@ Behavior after lamination:
...
+      :ref:`unifyfs utility <unifyfs_utility_label>` with its data staging
+      options.
 
-    - Can be used with checkpointing libraries (VeloC) or I/O libraries to
-      support shared files on burst buffers
+    - UnifyFS also can be coupled with SymphonyFS_, high level I/O libraries, or
```
Comment (CamStan) [inline docs/assumptions.rst:130]:
```suggestion
    - UnifyFS also can be coupled with SymphonyFS_, high level I/O libraries, or a
```

**Comment by sandrain** [comment]
Thank you, @CamStan . I've fixed the problems.

**Comment by sandrain** [comment]
As we discussed, the fsync() is now removed from the example. @adammoody @CamStan

**Comment by adammoody** [comment]
Thanks @sandrain and @CamStan

**Pull request merged**